### PR TITLE
Add watch verb to snapshot.storage.k8s.io group

### PR DESCRIPTION
**Pull request by OdedViner**
https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storagecluster/storagecluster_controller.go#L232

### CODE REVIEW

File: controllers/storagecluster/reconcile.go [hunk 1]
```diff
@@ -114,7 +114,7 @@ var validTopologyLabelKeys = []string{
 // +kubebuilder:rbac:groups=core,resources=namespaces,verbs=get
 // +kubebuilder:rbac:groups=apps,resources=deployments;daemonsets;replicasets;statefulsets,verbs=get;list;watch;create;update;delete
 // +kubebuilder:rbac:groups=monitoring.coreos.com,resources=servicemonitors;prometheusrules,verbs=get;list;watch;create;update;delete
-// +kubebuilder:rbac:groups=snapshot.storage.k8s.io,resources=volumesnapshots;volumesnapshotclasses,verbs=get;create;delete
+// +kubebuilder:rbac:groups=snapshot.storage.k8s.io,resources=volumesnapshots;volumesnapshotclasses,verbs=get;watch;create;delete
```
Comment (Madhu-1) [inline controllers/storagecluster/reconcile.go:117]: AFAIK ocs-operator doesnt need volumesnapshots permission that can be removed and also you need update access for volumesnapshotclass

Comment (OdedViner) [reply controllers/storagecluster/reconcile.go]: https://github.com/red-hat-storage/ocs-operator/pull/2774#issuecomment-2325063959
@Madhu-1 I don't find the `update` verb in the code.
Removed `volumesnapshots` resource

Comment (Madhu-1) [reply controllers/storagecluster/reconcile.go]: https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storagecluster/volumesnapshotterclasses.go#L132 here we are doing update

**Review by iamniting — CHANGES_REQUESTED** [review]
We should not remove volumesnapshots permissions

**Comment by iamniting** [comment]
> We should not remove volumesnapshots permissions

Discard, Missed Madhu's comment.

File: controllers/storagecluster/reconcile.go (hunk 1 above)
Comment (OdedViner) [reply controllers/storagecluster/reconcile.go]: @Madhu-1 @iamniting done

**Pull request merged**